salt/pillar/netbox.py

```diff
@@ -1,129 +1,198 @@
-
-Create a token in your NetBox instance at
-http://netbox_url.com/user/api-tokens/
-
-By default, this module will query the NetBox API for the platform associated
-with the device, and use the 'NAPALM driver' field to set the napalm
-proxy-minion driver. (Currently only 'napalm' is supported for drivertype.)
-
-This module assumes you will use SSH keys to authenticate to the network device
-If password authentication is desired, it is recommended to create another
-``proxy`` key in pillar_roots (or git_pillar) with just the ``passwd`` key and
-use :py:func:`salt.renderers.gpg <salt.renderers.gpg>` to encrypt the value.
-If any additional options for the proxy setup are needed they should also be
-configured in pillar_roots.
+    proxy_auth:
+      router1_username: router1
+      router1_password: password
+      junos_username: junos
+      junos_password: password
+      username: username
+      password: password
 '''
 
-
 from __future__ import absolute_import, print_function, unicode_literals
+
 import logging
 
+log = logging.getLogger(__name__)
+
 try:
-    import requests
-    import ipaddress
-    _HAS_DEPENDENCIES = True
+    import pynetbox
+    HAS_PYNETBOX = True
 except ImportError:
-    _HAS_DEPENDENCIES = False
+    HAS_PYNETBOX = False
+    log.error('pynetbox must be installed')
 
-log = logging.getLogger(__name__)
+try:
+    import netaddr
+    HAS_NETADDR = True
+except ImportError:
+    HAS_NETADDR = False
+    log.error('netaddr must be installed')
+
+# Pull the following from Netbox
+DEVICE_FIELDS = [
+    'id',
+    'display_name',
+    'device_type',
+    'device_role',
+    'primary_ip',
+    'primary_ip4',
+    'primary_ip6',
+    'custom_fields'
+]
+
+INTERFACE_FIELDS = [
+    'id',
+    'name',
+    'enabled',
+    'mtu',
+    'mac_address',
+    'description',
+    'form_factor',
+    'is_connected',
+    'mgmt_only',
+    'lag'
+]
+
+IP_FIELDS = [
+    'id',
+    'address',
+    'role',
+    'description',
+    'tenant',
+    'vrf.name',
+    'vrf.rd'
+]
 
 
 def __virtual__():
-    return _HAS_DEPENDENCIES
+    if HAS_PYNETBOX and HAS_NETADDR:
+        return True
+    return False
 
 
-def ext_pillar(minion_id, pillar, *args, **kwargs):
+def ext_pillar(minion_id,
+               pillar,  # pylint: disable=W0613
+               url,
+               token=None,
+               keyfile=None,
+               proxy_config=False):
     '''
-    Query NetBox API for minion data
+    Get pillar information from Netbox
     '''
 
-    # Pull settings from kwargs
-    api_url = kwargs['api_url'].rstrip('/')
-
-    api_token = kwargs.get('api_token', None)
-    proxy_username = kwargs.get('proxy_username', None)
-    proxy_return = kwargs.get('proxy_return', True)
-
+    nb_api = pynetbox.api(url, token=token, private_key_file=keyfile)
+    device = nb_api.dcim.devices.get(name=minion_id)
+    device_info = {}
+    if device:
+        device.platform.full_details()
+        napalm_driver = device.platform.napalm_driver
```

Comment:
I think these should be wrapped in a `try` and set napalm_driver to None on exception. This will handle the case of platform being None. (The NetBox interface doesn't require platform, and in my test case I hadn't set it, so I got an exception here.)